### DISCUSSION TITLE
Better stats for multi-station rides

### DISCUSSION THREAD

**Discussion by jensj12** [OpenRCT2/OpenRCT2#15962] [Feature requests]
(inspired by #15953)
Reasoning: Rides with multiple stations (not considering transport rides now, think of möbius coasters) have a few problems. Since most stats are ride-wide, there is no way to distinguish gentle and wild parts of the ride, and these rides tend to have higher stats (especially with inversions). The calculated ratings apply to the whole ride, regardless of the part the guest actually rides. This can make the ride very intense even if the individual parts would not be, or make guests willing to pay a lot of money for just a tiny part of the ride (which I would consider broken). As a bonus, since NSF, the stats window can't handle more than 4 stations and needs to be changed anyways.

Proposed solution: All stats are calculated separately for every station (or every for every entrance until the next exit). The stats window gets a dropdown to select the part of the attraction to show stats from, with an extra option for total/average stats.

Stretch goals: Guests can select to station to head for, rather than just a ride (they currently pick the closest entrance once a ride is chosen). Think of it as each entrance being a different ride, but sometimes part of the same attraction. Being able to set different prices for different entrances would be a nice bonus.

Estimated technical work: The stats calculator would need to index all its stats (which is quite a lot) on stations (and save it properly), and the stats window would need a rework partly already required to fix #15953 properly.

**Comment by ocalhoun6** [comment]
> Stretch goals: Guests can select to station to head for, rather than just a ride (they currently pick the closest entrance once a ride is chosen). Think of it as each entrance being a different ride, but sometimes part of the same attraction. Being able to set different prices for different entrances would be a nice bonus.

This would also nicely solve #13499

**Reply by jensj12** [reply]
Thanks, I wanted to link that issue, but couldn't find it.

**Comment by YJSoft** [comment]
![image](https://user-images.githubusercontent.com/5893514/142995397-e8fe89ae-9f73-49f0-92b4-ae78a54e4b18.png)
Quick mockup design

**Reply by duncanspumpkin** [reply]
I would suggest you split it into section specific station stats, ratings and common stats.
Most users will only care about the common or station 1 stats.

**Reply by Broxzier** [reply]
I'd also call it 'section' instead of 'station'. The order of stations isn't always incremental, it depends on the order they have been built.

**Reply by toimine** [reply]
also you could put it at the top instead, excitement/intensity/nausea are the stats you care the most about so it would be cool to have both them and the section number snugly inside one's field of view
